mentat/parsers/git_parser.py

```diff
@@ -23,11 +39,33 @@ async def stream_and_parse_llm_response(
         self,
         response: AsyncGenerator[Any, None],
     ) -> ParsedLLMResponse:
-        string = ""
+        current_line = ""
+        conversation = ""
+        file_edits = ""
         async for chunk in response:
             for content in chunk_to_lines(chunk):
-                string += content
-        return self.parse_string(string)
+                # Wait for a full line
```

Comment:
Why are we parsing it this way? If we aren't streaming it, this just adds a bunch of complexity for no reason.